File: Q68/Q68.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def in_order_traversal(root):
    stack = []  
    current = root

    while current is not None or stack:
        
        while current is not None:
            stack.append(current)
            current = current.left

        current = stack.pop()
        print(current.data, end=" ")  
      
        current = current.right

if __name__ == "__main__":
  
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)

    print("In-order traversal:", end=" ")
    in_order_traversal(root)

File: Q68/Q68.cpp
#include <iostream>
#include <stack>
using namespace std;

struct Node 
{
    int data;
    Node *left, *right;

    Node(int value) 
{ 
        data = value;
        left = right = nullptr;
    }
};

void inOrderTraversal(Node* root) 
{
    stack<Node*> s; 
    Node* current = root;

    while (current != nullptr || !s.empty()) 
    {    
        while (current != nullptr) 
        {
            s.push(current);
            current = current->left;
        }

        current = s.top();
        s.pop();
        cout << current->data << " "; 

        current = current->right;
    }
}

int main() 
{
    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);

    cout << "In-order traversal: ";
    inOrderTraversal(root);

    return 0;
}
